app/manager/models.py

```diff
@@ -18,163 +19,126 @@ def __str__(self):
 
     def __str__(self):
-        """String for representing"""
         return self.name
 
 
-class Cluster(models.Model):
-    """Model class for defining a cluster"""
+class HostStatus(models.Model):
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
 
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-    env = models.ForeignKey(EnvType, on_delete=models.CASCADE)
+    def __str__(self):
+        return self.name
 
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+
+class Service(models.Model):
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
 
     def __str__(self):
-        """String for representing"""
         return self.name
 
 
 class Group(models.Model):
-    """Model class for defining groups"""
-
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-    cluster = models.ForeignKey(Cluster, on_delete=models.CASCADE)
-
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
 
     def __str__(self):
-        """String for representing"""
         return self.name
 
 
 class Host(models.Model):
-    """Model class for defining a host"""
+    hostname = models.CharField(max_length=128, null=False, blank=False, unique=True)
+    fqdn_hostname = models.CharField(max_length=128, null=False, blank=False, unique=True)
+    generation = models.CharField(max_length=64, null=True, blank=True)
+    services = models.ManyToManyField(Service)
+    groups = models.ManyToManyField(Group)
+    cpu_nb = models.IntegerField(null=False, blank=False)
+    cpu_speed = models.FloatField(null=False, blank=False)
+    cores_nb = models.IntegerField(null=False, blank=False)
+    threads_nb = models.IntegerField(null=False, blank=False)
+    gpu_speed = models.FloatField(null=False, blank=False)
+    gpu_nb = models.IntegerField(null=False, blank=False)
+    cluster = models.ForeignKey(Cluster, null=False, blank=False, on_delete=models.PROTECT)
+    host_type = models.ForeignKey(HostType, null=False, blank=False, on_delete=models.PROTECT)
+    host_status = models.ForeignKey(HostStatus, null=False, blank=False, on_delete=models.PROTECT)
 
-    # Fields
-    hostname = models.CharField(max_length=128, null=False)
-    fqdn_hostname = models.CharField(max_length=128, null=False)
-    server = models.ForeignKey(Server, on_delete=models.CASCADE)
-    cluster = models.ForeignKey(Cluster, on_delete=models.CASCADE)
-    host_type = models.ForeignKey(HostType, on_delete=models.CASCADE)
-    host_status = models.ForeignKey(HostStatus, on_delete=models.CASCADE)
-    group = models.ForeignKey(Group, on_delete=models.CASCADE)
-    service = models.ManyToManyField(Service)
-    rack = models.CharField(max_length=64, null=False)
-    rack_position = models.PositiveSmallIntegerField()
```

Comment:
We have to keep these 2 fields as they are important to have a idea of where are the hosts in the racks and we use it for the rack toplogy configuration

Reply:
Done :)